Auto complete highlghter

Martinesp opened the pull request:
# PR Checklist

## What type of change is this?
- [ ] Internal (Does not need to be published)
- [ ] Patch (No new feature and no breaking change)
- [X ] Minor (New feature, but non breaking)
- [ ] Major (Breaking change)

## Have you bumped the package version?
- [X ] Yes
- [ ] No, does not need to be published

## Does this PR relate to an open issue?
- [ ] Yes (remember to link)
- [X ] No

## Have you added/updated the necessary tests?
- [X] Yes
- [ ] No, not needed

## Describe PR briefly:

Added Highlighter pipe to bold autocomplete options that matches the letters in the searchstring.

Martinesp committed:
improvement(autocomplete): inital start of highlighter pipes

Martinesp committed:
Highlither function

Martinesp committed:
Merge branch 'AutoComplete-improvements' into autoComplete-highlghter

Martinesp committed:
Update autocomplete.component.html

Martinesp committed:
improvement(highlither): temporary solution. Needs more work

Martinesp committed:
improvement(autocomplete): added working highlighter, bolds letters that match

Martinesp committed:
Update autocomplete.component.spec.ts

Martinesp committed:
Update autocomplete.component.ts

Martinesp committed:
Merge branch 'trunk' into autoComplete-highlghter

Martinesp committed:
Update package.json

Martinesp committed:
test(autocomplete pipe): added test for highligh pipe

Martinesp committed:
Update highlight-autocomplete.pipe.spec.ts

Martinesp committed:
Update highlight-autocomplete.pipe.spec.ts

frideska reviewed (changes requested):
Ser bra ut, fant en ting som fungerer litt rart: Hvis man prøver å velge et alternativ før man har søkt ferdig vil ikke selectoren velge alterativet ved klikk alltid, noen ganger fungerer det og andre ganger ikke! (funker fint med pil-taster)!

JGH153 commented on projects/hal-components/src/lib/components/autocomplete/highlight-autocomplete.pipe.spec.ts:
```diff
@@ -0,0 +1,16 @@
+    expect(pipe).toBeTruthy();
+  });
+
+  it('returns option with matching search term in bold', () => {
```
Lag en negativ test også + en som matcher to ganger

Martinesp committed:
test(autocomplete): added test for autocomplete Highlighter pipe

Martinesp committed:
Merge branch 'trunk' into autoComplete-highlghter

Martinesp committed:
fix(autocomplete): updated package version, and increased timer on blur to fix bug

Sometimes the option did not get selected when you selected an item from the list in the middle of a
search

Martinesp committed:
test(autocomplete): updated test

Martinesp committed:
Merge branch 'trunk' into autoComplete-highlghter
